server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -444,6 +450,20 @@ public long sizeOfGensAboveSeqNoInBytes(long minSeqNo) {
         }
     }
 
+    private long earliestGenerationLastModifiedAge(long minGeneration) {
+        try (ReleasableLock ignored = readLock.acquire()) {
+            ensureOpen();
+            Optional<? extends BaseTranslogReader> earliestGeneration = Stream.concat(readers.stream(), Stream.of(current))
+                .filter(r -> r.getGeneration() >= minGeneration)
+                .min(Comparator.comparingLong(BaseTranslogReader::getGeneration));
+            try {
+                return earliestGeneration.isPresent() ? earliestGeneration.get().getLastModifiedTime() : 0L;
```

Comment:
the method indicates we return an age by this returns last modified time?

Reply:
Derp.